[tensorexpr] support for fusing autocasting ops

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72478

aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

davidberard98 committed:
[tensorexpr] support for fusing autocasting ops

aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

davidberard98 commented:
RUN_TORCHBENCH: ALL

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

eellison commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -981,16 +996,23 @@ class TensorExprFuser {
           return false;
         }
       }
-      // cant support non-constant pin_memory or pin_memory = True
```
Let's add the autocast nodes here: https://github.com/pytorch/pytorch/blob/95e6722e852d9d13f528945648b5ac6f086983c1/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L622

eellison reviewed (approved):
LGTM with one change

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

davidberard98 committed:
Update on "[tensorexpr] support for fusing autocasting ops"


aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).
